common/tree/script.go

```diff
@@ -418,26 +407,37 @@ func (d *CSVMultisigClosure) Decode(script []byte) (bool, error) {
 		return false, fmt.Errorf("empty script")
 	}
 
-	csvIndex := bytes.Index(
-		script, []byte{txscript.OP_CHECKSEQUENCEVERIFY, txscript.OP_DROP},
-	)
-	if csvIndex == -1 || csvIndex == 0 {
+	tokenizer := txscript.MakeScriptTokenizer(0, script)
+
+	// Advance the tokenizer to the next token and extract the sequence.
+	// If the opcode represents a small integer, store it as a single byte in the sequence.
+	// Otherwise, extract the data directly from the tokenizer.
+	if !tokenizer.Next() {
 		return false, nil
 	}
 
-	sequence := script[:csvIndex]
-	if len(sequence) > 5 {
-		return false, fmt.Errorf("invalid sequence length")
+	var sequence []byte
+	if txscript.IsSmallInt(tokenizer.Opcode()) {
+		sequence = []byte{tokenizer.Opcode()}
+	} else {
+		sequence = tokenizer.Data()
+	}
+
+	for _, opCode := range []byte{txscript.OP_CHECKSEQUENCEVERIFY, txscript.OP_DROP} {
+
+		if !tokenizer.Next() || tokenizer.Opcode() != opCode {
+			return false, nil
+		}
+
 	}
-	sequence = sequence[1:]
 
 	locktime, err := common.BIP68DecodeSequence(sequence)
 	if err != nil {
 		return false, err
 	}
 
 	multisigClosure := &MultisigClosure{}
-	valid, err := multisigClosure.Decode(script[csvIndex+2:])
+	valid, err := multisigClosure.Decode(tokenizer.Script()[tokenizer.ByteIndex():])
```

Comment:
would be better to put `tokenizer.Script()[tokenizer.ByteIndex():]` into a variable like:
```suggestion
	buf := tokenizer.Script()[tokenizer.ByteIndex():]
	valid, err := multisigClosure.Decode(buf)
```